src/main/java/kitchenpos/application/OrderService.java

```diff
@@ -1,109 +1,72 @@
 import kitchenpos.domain.OrderLineItem;
 import kitchenpos.domain.OrderStatus;
 import kitchenpos.domain.OrderTable;
+import kitchenpos.dto.OrderRequest;
+import kitchenpos.dto.OrderResponse;
+import kitchenpos.repository.MenuRepository;
+import kitchenpos.repository.OrderRepository;
+import kitchenpos.repository.OrderTableRepository;
 
 @Service
+@Transactional(readOnly = true)
 public class OrderService {
-	private final MenuDao menuDao;
-	private final OrderDao orderDao;
-	private final OrderLineItemDao orderLineItemDao;
-	private final OrderTableDao orderTableDao;
-
-	public OrderService(
-		final MenuDao menuDao,
-		final OrderDao orderDao,
-		final OrderLineItemDao orderLineItemDao,
-		final OrderTableDao orderTableDao
-	) {
-		this.menuDao = menuDao;
-		this.orderDao = orderDao;
-		this.orderLineItemDao = orderLineItemDao;
-		this.orderTableDao = orderTableDao;
+	private final MenuRepository menuRepository;
+	private final OrderRepository orderRepository;
+	private final OrderTableRepository orderTableRepository;
+
+	public OrderService(MenuRepository menuRepository, OrderRepository orderRepository,
+		OrderTableRepository orderTableRepository) {
+		this.menuRepository = menuRepository;
+		this.orderRepository = orderRepository;
+		this.orderTableRepository = orderTableRepository;
 	}
 
 	@Transactional
-	public Order create(final Order order) {
-		final List<OrderLineItem> orderLineItems = order.getOrderLineItems();
-
-		if (CollectionUtils.isEmpty(orderLineItems)) {
-			throw new IllegalArgumentException("주문 항목을 구성해야 주문이 가능합니다.");
-		}
-
-		final List<Long> menuIds = orderLineItems.stream()
-			.map(OrderLineItem::getMenuId)
-			.collect(Collectors.toList());
-
-		if (orderLineItems.size() != menuDao.countByIdIn(menuIds)) {
-			throw new IllegalArgumentException("주문 항목과 메뉴에서 찾은 주문항목의 수가 일치하지 않습니다.");
-		}
-
-		final OrderTable orderTable = orderTableDao.findById(order.getOrderTableId())
-			.orElseThrow(() -> new IllegalArgumentException("id에 해당하는 주문 테이블을 찾을 수 없습니다."));
-
-		if (orderTable.isEmpty()) {
-			throw new IllegalArgumentException("비어있는 테이블은 주문할 수 없습니다.");
-		}
-
-		order.setOrderTableId(orderTable.getId());
-		order.setOrderStatus(OrderStatus.COOKING.name());
-		order.setOrderedTime(LocalDateTime.now());
-
-		final Order savedOrder = orderDao.save(order);
-
-		final Long orderId = savedOrder.getId();
-		final List<OrderLineItem> savedOrderLineItems = new ArrayList<>();
-		for (final OrderLineItem orderLineItem : orderLineItems) {
-			orderLineItem.setOrderId(orderId);
-			savedOrderLineItems.add(orderLineItemDao.save(orderLineItem));
-		}
-		savedOrder.setOrderLineItems(savedOrderLineItems);
-
-		return savedOrder;
+	public OrderResponse create(final OrderRequest orderRequest) {
+		List<OrderLineItem> orderLineItems = createOrderLineItems(orderRequest);
+		final OrderTable orderTable = findOrderTable(orderRequest);
+		Order order = new Order(orderTable, OrderStatus.COOKING, LocalDateTime.now(), orderLineItems);
```

Comment:
```suggestion
		Order order = Order.ofCooking(orderTable, orderLineItems);
```
정적 팩터리 메서드를 추가해도 괜찮겠네요

Reply:
다음 단계에서 반영하도록하겠습니다 😄